Clarify version mismatch docs (#3246)
Particular/ServiceControl: docs/packaging.md

  - `/ServiceControl.Audit`
  - `/ServiceControl.Monitoring`

## The mechanics

The `ServiceControlInstaller.Packaging` project is responsible to output to a `deploy` folder in the root all the required artifacts for `ServiceControl`, `ServiceControl.Audit`, and `ServiceControl.Monitoring`. Those artifacts are later needed by:

- the `ServiceControlInstaller.Engine` project to create the above-mentioned required zip files
- the `Particular.PlatformSample.ServiceControl` project to create the Platform sample required NuGet package
- The Docker `dockerfile`(s) to build the Docker container images

## Assembly version mismatches

There can be an issue when the main instance folder and the selected transport each have a copy of the same assembly. This can happen when the transport/persistence package and the instance each reference the same dependency, but select different versions. At install time, one version or the other will be copied into the instance binary folder and things may break unexpectedly at runtime.

> To prevent this, the unit test `DeploymentPackageTests.DuplicateAssemblyShouldHaveMatchingVersions` tests the generated zip files for duplicated assemblies. If their versions match, then the test passes. If not then the test will fail with:

Reviewer: ```suggestion
To prevent this, the unit test `DeploymentPackageTests.DuplicateAssemblyShouldHaveMatchingVersions` tests if duplicated assemblies might be deployed. If their versions match, then the test passes. If not then the test will fail with:
```

```

> Microsoft.Bcl.AsyncInterfaces.dll has a different version in Instance/ServiceControl.Audit compared to Transports/AzureServiceBus

> This is telling you that the mismatch is happing in the ServiceControl.Audit and that the instance it self is clashing with the Azure ServiceBus transport and RavenDB 5 persister.

Reviewer: ```suggestion
This tells that the mismatch is happening in ServiceControl.Audit and that the instance itself is clashing with the Azure ServiceBus transport and RavenDB 5 persister.
```
